chain-abci/src/enclave_bridge/edp/server.rs

```diff
@@ -146,10 +145,20 @@ impl<T: EnclaveProxy> TxValidationServer<T> {
                         EnclaveResponse::UnknownRequest
                     }
                 };
-                let response = resp.encode();
-                self.socket
-                    .send(response, FLAGS)
-                    .expect("reply sending failed");
+                let mut response = resp.encode();
+                let mut response_len = response.len();
+                if response_len > (u32::MAX as usize) {
+                    log::error!("too large response: {}", response_len);
+                    response = EnclaveResponse::UnknownRequest.encode();
+                    response_len = response.len();
+                }
+                let response_len: [u8; 4] = u32::to_le_bytes(response_len as u32);
+                self.socket_to_enclave
+                    .write_all(&response_len)
+                    .expect("writing back tx-query response len");
+                self.socket_to_enclave
+                    .write_all(&response)
+                    .expect("writing back tx-query response");
```

Comment:
Can we use `Codec` trait in `enclave-protocol` for this?

Reply:
which `Codec` trait do you have in mind? the scale codec one or the tls one?
but yes, these things can be improved that it's decoded/encoded directly from/to the stream instead of doing this extra step if possible -- could be a part of https://github.com/crypto-com/chain/issues/2051

Reply:
Sorry, I meant traits in `codec.rs` (https://github.com/crypto-com/chain/blob/master/enclave-protocol/src/codec.rs)